### PR TITLE
Update_StackExchange_Redis_to_Version_2_2_62

### PR DESCRIPTION
**Pull request by alex-zaikman**

### CODE REVIEW

File: StackExchange.Redis.Resilience/StackExchange.Redis.Resilience.csproj [hunk 1]
```diff
@@ -3,11 +3,11 @@
   <PropertyGroup>
     <Product>StackExchange.Redis.Resilience</Product>
     <Title>StackExchange.Redis.Resilience</Title>
-    <Version>1.0.0</Version>
+    <Version>2.0.0</Version>
     <Authors>maca88</Authors>
     <Company />
     <Description>Provides a solution to force reconnect when StackExchange.Redis fails to reconnect after a connection blip.</Description>
-    <TargetFrameworks>net472;net461;netstandard2.0</TargetFrameworks>
+    <TargetFrameworks>net472;net461;netstandard2.0;net462</TargetFrameworks>
```
Comment (maca88) [inline StackExchange.Redis.Resilience/StackExchange.Redis.Resilience.csproj]: Why did you add `net462`? I think it would make more sense to align it with [StackExchange.Redis](https://github.com/StackExchange/StackExchange.Redis/blob/main/src/StackExchange.Redis/StackExchange.Redis.csproj#L4) `TargetFrameworks`.

File: StackExchange.Redis.Resilience.Tests/StackExchange.Redis.Resilience.Tests.csproj [hunk 2]
```diff
@@ -1,7 +1,7 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk">
 
   <PropertyGroup>
-    <TargetFramework>net472</TargetFramework>
+    <TargetFramework>net462</TargetFramework>
```
Comment (maca88) [inline StackExchange.Redis.Resilience.Tests/StackExchange.Redis.Resilience.Tests.csproj]: Why was this changed?

File: StackExchange.Redis.Resilience/ResilientDatabase.cs [hunk 3]
```diff
@@ -366,6 +371,12 @@ public Task<long> HashLengthAsync(RedisKey key, CommandFlags flags = CommandFlag
             return ExecuteActionAsync(() => _database.Value.HashLengthAsync(key, flags));
         }
 
+        public IAsyncEnumerable<HashEntry> HashScanAsync(RedisKey key, RedisValue pattern = new RedisValue(), int pageSize = 250, long cursor = 0,
+            int pageOffset = 0, CommandFlags flags = CommandFlags.None)
+        {
+            return ExecuteAction(() => _database.Value.HashScanAsync(key, pattern, pageSize, cursor));
```
Comment (maca88) [inline StackExchange.Redis.Resilience/ResilientDatabase.cs]: `pageOffset ` and `flags` parameters should also be passed.

File: StackExchange.Redis.Resilience/ResilientDatabase.cs [hunk 4]
```diff
@@ -1488,6 +1529,18 @@ public Task<long> SortedSetRemoveRangeByValueAsync(RedisKey key, RedisValue min,
             return ExecuteActionAsync(() => _database.Value.SortedSetRemoveRangeByValueAsync(key, min, max, exclude, flags));
         }
 
+        public IAsyncEnumerable<RedisValue> SetScanAsync(RedisKey key, RedisValue pattern = new RedisValue(), int pageSize = 250, long cursor = 0,
+            int pageOffset = 0, CommandFlags flags = CommandFlags.None)
+        {
+            return ExecuteAction(() => _database.Value.SetScanAsync(key, pattern, pageSize, cursor));
```
Comment (maca88) [inline StackExchange.Redis.Resilience/ResilientDatabase.cs]: `pageOffset ` and `flags` parameters should also be passed.

File: StackExchange.Redis.Resilience/ResilientServer.cs [hunk 5]
```diff
@@ -39,15 +39,21 @@ internal class ResilientServer : IServer
         public bool IsConnected => _server.Value.IsConnected;
 
         /// <inheritdoc />
+        [Obsolete("Starting with Redis version 5, Redis has moved to 'replica' terminology. Please use IsReplica instead.")]
         public bool IsSlave => _server.Value.IsSlave;
 
+        public bool IsReplica { get; }
```
Comment (maca88) [inline StackExchange.Redis.Resilience/ResilientServer.cs]: The getter should be changed to `=> _server.Value.IsReplica;`

File: StackExchange.Redis.Resilience/ResilientServer.cs [hunk 6]
```diff
@@ -39,15 +39,21 @@ internal class ResilientServer : IServer
...
+
         /// <inheritdoc />
+        [Obsolete("Starting with Redis version 5, Redis has moved to 'replica' terminology. Please use IsReplica instead.")]
         public bool AllowSlaveWrites
         {
             get => _server.Value.AllowSlaveWrites;
             set => _server.Value.AllowSlaveWrites = value;
         }
 
+        public bool AllowReplicaWrites { get; set; }
```
Comment (maca88) [inline StackExchange.Redis.Resilience/ResilientServer.cs]: The getter and setters should be changed to:
```
get => _server.Value.AllowReplicaWrites;
set => _server.Value.AllowReplicaWrites = value;
```

File: StackExchange.Redis.Resilience.Tests/StackExchange.Redis.Resilience.Tests.csproj (hunk 2 above)
Comment (alex-zaikman) [reply StackExchange.Redis.Resilience.Tests/StackExchange.Redis.Resilience.Tests.csproj]: My mistake, reverted.

File: StackExchange.Redis.Resilience/StackExchange.Redis.Resilience.csproj (hunk 1 above)
Comment (alex-zaikman) [reply StackExchange.Redis.Resilience/StackExchange.Redis.Resilience.csproj]: You are absolutly right, fixed.

File: StackExchange.Redis.Resilience/StackExchange.Redis.Resilience.csproj [hunk 7]
```diff
@@ -22,7 +22,7 @@
   </PropertyGroup>
 
   <ItemGroup>
-    <PackageReference Include="StackExchange.Redis" Version="[2.0.495,2.1)" />
+    <PackageReference Include="StackExchange.Redis"  Version="2.2.62" />
```
Comment (maca88) [inline StackExchange.Redis.Resilience/StackExchange.Redis.Resilience.csproj]: I would change this to `[2.2.3,2.3)` as it seems that `StackExchange.Redis` introduces breaking changes with minor versions.

File: StackExchange.Redis.Resilience/StackExchange.Redis.Resilience.csproj [hunk 8]
```diff
@@ -3,11 +3,11 @@
   <PropertyGroup>
     <Product>StackExchange.Redis.Resilience</Product>
     <Title>StackExchange.Redis.Resilience</Title>
-    <Version>1.0.0</Version>
+    <Version>2.0.0</Version>
```
Comment (maca88) [inline StackExchange.Redis.Resilience/StackExchange.Redis.Resilience.csproj:6]: I think that we should follow the `StackExchange.Redis` versioning and change this to `2.2.0`.

Comment (alex-zaikman) [reply StackExchange.Redis.Resilience/StackExchange.Redis.Resilience.csproj:6]: This is the version of your nuget and represents its changes.
A specific version number is in the form Major.Minor.Patch[-Suffix]
I would sugest folowing this convention, however, it is your decision :)

Comment (maca88) [reply StackExchange.Redis.Resilience/StackExchange.Redis.Resilience.csproj:6]: Ok, let's keep it as it is :)

**Review by maca88 — APPROVED** [review]

**Comment by maca88** [comment]
Thanks for the PR, the package was published so it is ready to be used :)

**Comment by alex-zaikman** [comment]
Awesome np :)

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________
